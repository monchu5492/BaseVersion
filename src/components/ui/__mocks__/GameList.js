const GameListMock = () => <div />;

GameListMock.displayName = 'GameListMock';

export default GameListMock;
