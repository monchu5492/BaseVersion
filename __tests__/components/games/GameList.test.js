import toJSON from 'enzyme-to-json';
import { shallow, mount } from 'enzyme';
import { compose } from 'redux';
import GameList from '../../../src/components/ui/GameList';

jest.mock('../../../src/components/ui/Game', () =>
  ({ onRemove = f => f }) =>
    (<div className="mock-game">
      <button className="remove" onClick={onRemove} />
     </div>));

describe('<GameList /> UI Component', () => {
  afterAll(() => jest.resetAllMocks());

  describe('Rendering UI', () => {
    it('Renders Correctly', () =>
      compose(expect, toJSON, shallow)(<GameList games={_testGames} />).toMatchSnapshot());

    it('Defaults Properties correctly', () =>
      expect(shallow(<GameList />).find('p').text())
        .toBe('No Games Listed. (Add a Game)'));
  });
});
