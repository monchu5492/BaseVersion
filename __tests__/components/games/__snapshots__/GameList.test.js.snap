// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GameList /> UI Component Rendering UI Renders Correctly 1`] = `
<div
  className="game-list"
>
  <Component
    color="#ff4b47"
    id="58d9caee-6ea6-4d7b-9984-65b145031979"
    key="58d9caee-6ea6-4d7b-9984-65b145031979"
    onRate={[Function]}
    onRemove={[Function]}
    rating={0}
    timestamp="Fri Apr 15 2016 12:54:43 GMT-0700 (PDT)"
    title="tomato"
  />
</div>
`;
